File: 4.counting_elements/lesson_4_perm_check.py
def solution(A):
    maxi = 0
    for element in A:
        if(maxi < element):
            maxi = element
    if(maxi != len(A)):
        return 0
    occurs = [False] * maxi
    for element in A:
        occurs[element-1] = True
    if(False in occurs):
        return 0
    return 1

File: 4.counting_elements/lesson_4_perm_check.js
function solution(A) {
    let map = new Map()

    for(let i = 0; i < A.length; i++) {
        map.set(A[i], 1)
    }
    for(let i = 1; i <= A.length; i++) {
        if(!map.has(i)) {
            return 0
        }
    }
    return 1
}